src/pipx/main.py

```diff
@@ -605,6 +607,21 @@ def _add_ensurepath(subparsers: argparse._SubParsersAction) -> None:
     )
 
 
+def _add_environment(subparsers: argparse._SubParsersAction) -> None:
+    p = subparsers.add_parser(
+        "environment",
+        help=("Print a list of variables used in pipx.constants."),
+        description=(
+            "Available variables: "
+            "PIPX_HOME, PIPX_BIN_DIR, PIPX_SHARED_LIBS, PIPX_LOCAL_VENVS, PIPX_LOG_DIR, "
+            "PIPX_TRASH_DIR, PIPX_VENV_CACHEDIR"
+        ),
+    )
+    p.add_argument(
+        "--value", "-v", metavar="VARIABLE", help="Print the value of the variable."
+    )
+
+
```

Comment:
I think this is useful, but I am concerned there will be some confusion about whether these are settable by users or not. 

Can you add a note to the end of the actual output (in environment.py) mentioning only some of those are settable, and to see the list, see the output of `pipx --help`.

Reply:
I think only `PIPX_HOME` and `PIPX_BIN_DIR` are settable currently in the above list. Probably the note can display that they are settable directly, instead of telling users to run a new command?

Reply:
> I think only `PIPX_HOME` and `PIPX_BIN_DIR` are settable currently in the above list. Probably the note can display that they are settable directly, instead of telling users to run a new command?

Sure that works